frontend/apps/reader/modules/readertypeset.lua

```diff
@@ -428,14 +429,22 @@ function ReaderTypeset:addToMainMenu(menu_items)
 end
 
 function ReaderTypeset:makeDefaultFloatingPunctuation()
-    local toggler = self.floating_punctuation == 1 and _("On") or _("Off")
-    UIManager:show(ConfirmBox:new{
+    local floating_punctuation = G_reader_settings:isTrue("floating_punctuation")
+    local toggler = floating_punctuation and _("Disable") or _("Enable")
+    UIManager:show(MultiConfirmBox:new{
         text = T(
-            _("Set default hanging punctuation to %1?"),
+            _("%1 hanging punctuation by default?"),
```

Comment:
That should be written out in full. There's no guarantee that enable and disable are interchangeable in other languages, let alone take the same form without context.

However, it's a moot point because it should take the same form as the others:

```
Would you like to enable or disable hanging punctuation by default? The current default (★) is enabled.
```

Reply:
i was trying to leave it as intact as possible, i didn't realize the implications.